src/lsli/main.py

```diff
@@ -0,0 +1,449 @@
+    )
+    cli_handler.setFormatter(formatter)
+
+    log_handler = logging.FileHandler(log_path, mode="w")
+    log_handler.setLevel(config.LOG_LEVEL)
+    log_handler.setFormatter(formatter)
+
+    module_logger.addHandler(cli_handler)
+    module_logger.addHandler(log_handler)
+    palletjack_logger.addHandler(cli_handler)
+    palletjack_logger.addHandler(log_handler)
+
+    #: Log any warnings at logging.WARNING
+    #: Put after everything else to prevent creating a duplicate, default formatter
+    #: (all log messages were duplicated if put at beginning)
+    logging.captureWarnings(True)
+
+    module_logger.debug("Creating Supervisor object")
+    skid_supervisor = Supervisor(handle_errors=False)
+    sendgrid_settings = config.SENDGRID_SETTINGS
+    sendgrid_settings["api_key"] = sendgrid_api_key
+    skid_supervisor.add_message_handler(
+        SendGridHandler(
+            sendgrid_settings=sendgrid_settings, client_name=config.SKID_NAME, client_version=version.__version__
+        )
+    )
+
+    return skid_supervisor
+
+
+def _remove_log_file_handlers(log_name, loggers):
+    """A helper function to remove the file handlers so the tempdir will close correctly
+
+    Args:
+        log_name (str): The logfiles filename
+        loggers (List<str>): The loggers that are writing to log_name
+    """
+
+    for logger in loggers:
+        for handler in logger.handlers:
+            try:
+                if log_name in handler.stream.name:
+                    logger.removeHandler(handler)
+                    handler.close()
+            except Exception:
+                pass
+
+
+def process():
+    """The main function that does all the work."""
+
+    #: Set up secrets, tempdir, supervisor, and logging
+    start = datetime.now()
+
+    secrets = SimpleNamespace(**_get_secrets())
+
+    with TemporaryDirectory() as tempdir:
+        tempdir_path = Path(tempdir)
+        log_name = f'{config.LOG_FILE_NAME}_{start.strftime("%Y%m%d-%H%M%S")}.txt'
+        log_path = tempdir_path / log_name
+
+        skid_supervisor = _initialize(log_path, secrets.SENDGRID_API_KEY)
+
+        #: Get our GIS object via the ArcGIS API for Python
+        gis = arcgis.gis.GIS(config.AGOL_ORG, secrets.AGOL_USER, secrets.AGOL_PASSWORD)
+
+        module_logger.info("Loading data from graphql endpoint...")
+        records_df = _load_records_from_graphql(secrets.GRAPHQL_URL, config.GRAPHQl_QUERY, config.GRAPHQL_LIMIT)
+
+        module_logger.info("Transforming data...")
+        spatial_records = _spatialize_data(records_df)
+        spatial_records.rename(
+            columns={"serviceline_material_cassification": "serviceline_material_cassificat"}, inplace=True
+        )
+
+        #: Strip off trailing digits for any zipcodes in ZIP+4 format
+        spatial_records["pws_zipcode"] = spatial_records["pws_zipcode"].astype(str).str[:5].astype("Int64")
```

Comment:
not that memory matters much here, but a 5 digit number can be represented by less bits in a int32